src/terminal/parser/InputStateMachineEngine.hpp

```diff
@@ -90,6 +92,41 @@ namespace Microsoft::Console::VirtualTerminal
             CursorBackTab = L'Z',
         };
 
+        enum CsiIntermediateCodes : wchar_t
+        {
+            MOUSE_SGR = L'<',
+        };
+
+        enum CsiEndCodes : wchar_t
+        {
+            MOUSE_DOWN = L'M',
+            MOUSE_UP = L'm',
+        };
+
+        enum CsiMouseButtonCodes : unsigned short
+        {
+            Left = 0,
+            Middle = 1,
+            Right = 2,
+            Released = 3,
+            ScrollForward = 4,
+            ScrollBack = 5,
+            Button6 = 6,
+            Button7 = 7,
+            Button8 = 8,
+            Button9 = 9,
+            Button10 = 10,
+            Button11 = 11,
+        };
+
+        enum CsiMouseModifierCodes : unsigned short

```

Comment:
Looks like this is only used for `Drag` - should maybe some of the rest of the code be refactored to use the other values too?